common/src/main/java/us/myles/ViaVersion/api/platform/ViaConnectionManager.java

```diff
@@ -0,0 +1,54 @@
+package us.myles.ViaVersion.api.platform;
+
+import us.myles.ViaVersion.api.data.UserConnection;
+import us.myles.ViaVersion.protocols.base.ProtocolInfo;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Handles UserConnections
+ */
+public class ViaConnectionManager {
+    protected final Map<UUID, UserConnection> clients = new ConcurrentHashMap<>();
+    protected final Set<UserConnection> connections = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    public void onLoginSuccess(UserConnection connection) {
+        Objects.requireNonNull(connection, "connection is null!");
+        UUID id = connection.get(ProtocolInfo.class).getUuid();
+        connections.add(connection);
+        clients.put(id, connection);
+    }
+
+    public void onDisconnect(UserConnection connection) {
+        Objects.requireNonNull(connection, "connection is null!");
+        UUID id = connection.get(ProtocolInfo.class).getUuid();
+        connections.remove(connection);
+        clients.remove(id);
+    }
+
+    /**
+     * Returns a map containing the UUIDs and frontend UserConnections from players connected to this proxy server
+     * Returns empty list when there isn't a server
+     */
+    public Map<UUID, UserConnection> getConnectedClients() {
+        return Collections.unmodifiableMap(clients);
+    }
+
+    /**
+     * Returns the frontend UserConnection from the player connected to this proxy server
+     * Returns null when there isn't a server or connection was not found
+     */
+    public UserConnection getConnectedClient(UUID clientIdentifier) {
+        return clients.get(clientIdentifier);
+    }
+
+    /**
+     * Returns all UserConnections which are active
+     * May contain duplicated UUIDs on multiple ProtocolInfo.
+     * May contain frontend, backend and/or client-sided connections.
```

Comment:
Might be worth noting that if they somehow do a /reload, the players won't be part of this list.